apps/avifenc.c

```diff
@@ -950,12 +950,10 @@ int main(int argc, char * argv[])
             gridCellIndex = gridCellCount - 1;
         }
 
-        for (uint32_t i = 0; i < gridCellCount; ++i) {
-            if (!gridCells[i]) {
-                fprintf(stderr, "ERROR: Not enough input files for grid image! (expecting %u, or a single image to be split)\n", gridCellCount);
-                returnCode = 1;
-                goto cleanup;
-            }
+        if (gridCellIndex != gridCellCount - 1) {
```

Comment:
By construction, gridCells[i] != NULL for i = 0 .. gridCellIndex. So if gridCellIndex != gridCellCount - 1, at least one gridCells[i] is NULL.